### ISSUE TITLE
Brew Plugin

### ISSUE THREAD

**Issue by hipstersmoothie** [intuit/auto#1098]
**Is your feature request related to a problem? Please describe.**

I want to automate my `brew` releases with an official `auto` plugin.

**Describe the solution you'd like**

A generic way to release `brew` packages.

**Describe alternatives you've considered**

`auto` currently uses it's own [custom plugin](https://github.com/intuit/auto/blob/master/scripts/update-tap.js) that accomplishes this.

**Comment by hipstersmoothie** [comment]
# Potential Solution

Users provide a `formula-template` that *is* their `brew` formula with variables for the `version` and `sha`.

```ruby
  class Auto < Formula
    desc "Generate releases based on semantic version labels on pull requests."
    homepage "https://intuit.github.io/auto/home.html"
    url "https://github.com/intuit/auto/releases/download/v$VERSION/auto-macos.gz"
    sha256 "$SHA"
    def install
      libexec.install Dir["*"]
      bin.install libexec/"auto-macos"
      mv bin/"auto-macos", bin/"auto"
    end
    test do
      system bin/"auto", "--version"
    end
  end
```

**Comment by chetanddesai** [comment]
Is `version` optional in a brew formula? I noticed yours just has `url` and `sha256`.  e.g. https://github.com/intuit/ReplayZero/blob/master/Formula/replay-zero.rb#L6

We can easily enough add it in

```
class Auto < Formula
  desc "Generate releases based on semantic version labels on pull requests."
  homepage "https://intuit.github.io/auto/home.html"
  url "https://github.com/intuit/auto/releases/download/v$VERSION/auto-macos.gz"
  version "$VERSION"
  sha256 "$SHA"
  def install
    libexec.install Dir["*"]
    bin.install libexec/"auto-macos"
    mv bin/"auto-macos", bin/"auto"
  end
  test do
    system bin/"auto", "--version"
  end
end
```

**Comment by hipstersmoothie** [comment]
@chetanddesai Good catch. I'll add that in

**Comment by wtait1** [comment]
Homebrew does try to guess the version from the `url` if it can, for example it does show the version for `auto`
```
❯ brew info auto                                                                                                   100%  ▓▒░
intuit/auto/auto: stable 9.22.2
Generate releases based on semantic version labels on pull requests.
https://intuit.github.io/auto/home.html
/usr/local/Cellar/auto/9.21.0 (3 files, 61.2MB) *
  Built from source on 2020-03-20 at 13:12:09
From: https://github.com/intuit/auto/blob/master/Formula/auto.rb
```

I just prefer to include it explicitly, like in the formula for Replay Zero 😄

**Comment by adierkens** [comment]
:rocket: Issue was released in [`v9.23.0`](https://github.com/intuit/auto/releases/tag/v9.23.0) :rocket:
